utils/docker/Dockerfile.centos.8

```diff
@@ -12,29 +12,77 @@ LABEL maintainer="daos@daos.groups.io"
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# If a local distro repository is supplied, then we should use it
```

Comment:
Is this file still used in Jenkins builds at all?  If not then we may be better off if we simply remove all the REPO_URL code couldn't we?  Having lots of new, complex but unused or untested code isn't going to move us forward.

Reply:
CentOS 8 is still used and is becoming more important for future DAOS builds

The REPO_URL prefix is used for both accessing the local mirrors of the distro repositories and for accessing prebuilt RPMs for dependencies.

I have tested the REPO_APPSTREAM in a fork of this PR to make sure it works, but the functionality will be enabled in a future pipeline-lib PR that is currently being tested against the existing DAOS.

Reply:
CentOS 8 is important as you say, but is the Dockerfile for CentOS 8 used in Jenkins?  If it's not then it would be better to have a simpler Dockerfile rather than a bunch of code which is not used and therefore by definition not tested.

Reply:
It is inevitable that something newer is going to become our core test distro as EL7 fades into the sunset. It could be CentOS 8 or it could be something else.  so we should maintain this functionality for such a future move and not have to do additional work at that time to _resurrect_ it once a move is made.  We (the CI team) just don't have the resources to put things into the bin and then have to take them back out and make something of them when it's decided at some future point that oh, yes, that was needed.

Reply:
I think we also want to maintain the ability to pivot that "core distro" to any of our supported distros without having to re-add the quick-build capabilities to it at the time.  Maybe its the nomenclature "quick-build" that is distracting.  Maybe it should be (thought of as -- I'm not suggesting that this PR actually do any kind of renaming) rpm-build, since it has more utility than just making the build _quick_.